src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -159,28 +169,33 @@ internal bool TryAddWithoutValidation(HeaderDescriptor descriptor, IEnumerable<s
                 throw new ArgumentNullException(nameof(values));
             }
 
-            using (IEnumerator<string?> enumerator = values.GetEnumerator())
+            using IEnumerator<string?> enumerator = values.GetEnumerator();
```

Comment:
The common case is certainly `TryAddWithoutValidation(string, string)`, but if we do end up using the `IEnumerable` overload, it would always be `string[]` (since the source is `StringValues`).